src/main/java/org/spongepowered/api/event/item/inventory/EnchantItemEvent.java

```diff
@@ -0,0 +1,114 @@
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.spongepowered.api.event.item.inventory;
+
+import org.spongepowered.api.data.DataSerializable;
+import org.spongepowered.api.data.Transaction;
+import org.spongepowered.api.item.enchantment.Enchantment;
+import org.spongepowered.api.item.inventory.ItemStack;
+import org.spongepowered.api.item.inventory.Slot;
+
+import java.util.List;
+
+/**
+ * This event is triggered when an item is enchanted at an enchanting table.
+ */
+public interface EnchantItemEvent extends ClickInventoryEvent {
+
+    /**
+     * Returns the slot of the enchanted item.
+     *
+     * @return the slot if the enchanted item.
+     */
+    Slot getSlot();
+
+    /**
+     * Triggers when the enchantment costs for an item are calculated.
+     */
+    // Mixin onCraftMatrixChanged in loop after EnchantmentHelper.calcItemStackEnchantability
+    // Forge: ForgeEventFactory.onEnchantmentLevelSet
+    interface CalculateCost extends TargetContainerEvent {
+
+        int getSeed(); // pseudo random seed
+        int getSlot(); // 1,2,3
+        int getPower(); // based on books (0-?)
+
+        ItemStack getItem(); // item to enchant
+
+        // TODO Transaction that allows non DataSerializable types?
+        Transaction<EnchantmentCost> getCost();
+
+        interface EnchantmentCost extends DataSerializable {
+            int cost();
+            // vanilla max 30
+            // going over 30 a lot may break vanilla enchanting
+            EnchantmentCost withCost(int cost);
+        }
+
+    }
+
+    /**
+     * Triggers when the enchantments for an item are calculated.
+     * This happens after a new item to enchant is put in the table
+     * and again when the item is enchanted.
+     */
+    // Mixin ContaienrEnchantment.getEnchantmentList and/or
+    //       EnchantmentHelper.buildEnchantmentList-at-getEnchantmentDatas
+    interface CalculateEnchantment extends TargetContainerEvent {
+
+        // Seed + slot are seed for enchantment-list generation
+        int getSeed();
+        // 1,2,3
+        int getSlot();
+        // vanilla max 30
+        int getCost();
+
+        // First Enchantment in list is used for clue in client view
+        Transaction<EnchantmentList> getEnchantments();
+
+        interface EnchantmentList extends DataSerializable {
```

Comment:
I'd rather this get moved into it's own class/package where Enchantments resides.

Reply:
And what exactly is an enchantment list supposed to represent? Is it something closer to Data API's `Keys.ENCHANTMENTS`?

Reply:
It will get moved out. Just in the same file for visibility atm.

Using the `EnchantmentData` would probably be fine.
I wanted something where I could add utility methods to:
 - reroll random enchantments with different parameters
 - reroll random enchantments with fixed enchantment pool
 - set enchantment result 

move those to smth like `EnchantmentUtil` ?